src/javascript/_common/base/gtm.js

```diff
@@ -27,7 +26,8 @@ const GTM = (() => {
     });
 
     const pushDataLayer = (data) => {
-        if (isGtmApplicable() && !Login.isLoginPages()) {
+        const is_login_pages = /logged_inws|redirect/i.test(window.location.pathname);
```

Comment:
why don't we fix the function?

Reply:
i put the function back to normal. however. we can't import login here since it will be circular dependency here

Reply:
what is the circular dependency?

Reply:
as discussed, please move it to a utility function to eliminate the race condition.